mbed/mbed.py

```diff
@@ -607,8 +607,24 @@ def cleanup():
         for branch in branches: # delete all local branches so the new repo clone is not poluted
             pquery([git_cmd, 'branch', '-D', branch])
 
-    def clone(url, name=None, depth=None, protocol=None):
-        popen([git_cmd, 'clone', formaturl(url, protocol), name] + (['--depth', depth] if depth else []) + (['-v'] if very_verbose else ([] if verbose else ['-q'])))
+    def clone(url, path, rev=None, depth=None, protocol=None, name=None):
+        result = pquery([git_cmd, "ls-remote", url, (rev if rev else "HEAD")])
+        
+        if result and rev:
+            repo_name = url.split('/')[-1]
+            if '.git' in repo_name:
+                repo_name = repo_name[:-4]
+
+            os.mkdir(repo_name)
+
+            with cd(repo_name):
+                Git.init()
```

Comment:
Why not try to find an alternative approach instead of hacking it yourself? E.g.
1. Clone using the original depth command
2. Add an if statement if --depth is used to call fetch with revision/tag parameter.

Worst case scenario this would fetch the latest from master + latest tag, vs 2 calls git ls-remote + git fetch.

Reply:
This actually is the alternative approach. The entire goal of the feature was to download as little information as possible, to speed up the command.

To use `git clone` first would defeat the purpose of the enhancement, since downloading the entire repo history is excessive if the user only wants the files from a single release. I didn't find a flag for `git clone` that would allow for cloning a repo at a specific ref, hence the split.